tests/ui/argument-suggestions/extra_arguments.stderr

```diff
@@ -229,6 +248,33 @@ note: function defined here
 LL | fn two_arg_diff(_a: i32, _b: &str) {}
    |    ^^^^^^^^^^^^ -------  --------
 
-error: aborting due to 14 previous errors
+error[E0061]: this function takes 0 arguments but 2 arguments were supplied
+  --> $DIR/extra_arguments.rs:8:9
+   |
+LL |         empty($x, 1);
+   |         ^^^^^     - unexpected argument of type `{integer}`
+...
+LL |   foo!(1);
+   |   -------
+   |   |    |
+   |   |    unexpected argument of type `{integer}`
+   |   in this macro invocation
+   |
+note: function defined here
+  --> $DIR/extra_arguments.rs:1:4
+   |
+LL | fn empty() {}
+   |    ^^^^^
+   = note: this error originates in the macro `foo` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: remove the extra arguments
+   |
+LL ~         empty($x, );
```

Comment:
hmmm.. this isn't right still? maybe we should give up if the removal is within a macro?

Reply:
In the added commit I've removed adding the suggestion if it falls within a macro expansion

Reply:
Could you reapply this part tho? Or was it too entangled into the other changes you reverted?

Reply:
Added it back, reverted a bit too much